File: domain/src/test/java/com/arturogutierrez/openticator/domain/otp/time/RemainingTimeCalculatorTest.java
package com.arturogutierrez.openticator.domain.otp.time;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.mockito.Mockito.when;

public class RemainingTimeCalculatorTest {

  @Test
  public void testCalculateRemainingSeconds() {
    TimeProvider timeProvider = new StubTimeProvider(100L);
    RemainingTimeCalculator remainingTimeCalculator = new RemainingTimeCalculator(timeProvider);

    int remainingSeconds = remainingTimeCalculator.calculateRemainingSeconds(120L);

    assertThat(remainingSeconds, is(20));
  }

  @Test
  public void testRemainingSecondsShouldBeZeroIfCurrentTimeIsAfterValidUntilDate() {
    TimeProvider timeProvider = new StubTimeProvider(50L);
    RemainingTimeCalculator remainingTimeCalculator = new RemainingTimeCalculator(timeProvider);

    int remainingSeconds = remainingTimeCalculator.calculateRemainingSeconds(20L);

    assertThat(remainingSeconds, is(0));
  }
}

File: domain/src/test/kotlin/com/arturogutierrez/openticator/domain/otp/time/RemainingTimeCalculatorTest.kt
package com.arturogutierrez.openticator.domain.otp.time

import com.arturogutierrez.openticator.domain.otp.stubs.StubTimeProvider
import com.natpryce.hamkrest.assertion.assertThat
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class RemainingTimeCalculatorTest {

  @Test
  fun testCalculateRemainingSeconds() {
    val timeProvider = StubTimeProvider(100L)
    val remainingTimeCalculator = RemainingTimeCalculator(timeProvider)

    val remainingSeconds = remainingTimeCalculator.calculateRemainingSeconds(120L)

    assertThat(remainingSeconds, equalTo(20L))
  }

  @Test
  fun testRemainingSecondsShouldBeZeroIfCurrentTimeIsAfterValidUntilDate() {
    val timeProvider = StubTimeProvider(50L)
    val remainingTimeCalculator = RemainingTimeCalculator(timeProvider)

    val remainingSeconds = remainingTimeCalculator.calculateRemainingSeconds(20L)

    assertThat(remainingSeconds, equalTo(0L))
  }
}
